python/core/raster/qgsrasterdataprovider.sip

```diff
@@ -233,10 +233,44 @@ class QgsRasterDataProvider : QgsDataProvider, QgsRasterInterface
     /** Current time stamp of data source */
     virtual QDateTime dataTimestamp() const;
 
+    /** Checks whether the provider is in editing mode, i.e. raster write operations will be accepted.
+     * By default providers are not editable. Use setEditable() method to enable/disable editing.
+     * @see setEditable(), writeBlock()
+     * @note added in QGIS 3.0
+     */
+    virtual bool isEditable() const;
+
+    /** Turns on/off editing mode of the provider. When in editing mode, it is possible
+     * to overwrite data of the provider using writeBlock() calls.
+     * @note Only some providers support editing mode and even those may fail to turn turn
+     * the underlying data source into editing mode, so it is necessery to check afterwards
+     * with isEditable() whether the operation was successful.
+     * @see isEditable(), writeBlock()
+     * @note added in QGIS 3.0
+     */
+    virtual void setEditable( bool enabled );
```

Comment:
Yeah it would probably make sense... I wanted to keep the setter "clean" with no return value, but here a return value may be useful :-)